app/src/main/java/openfoodfacts/github/scrachx/openfood/views/splash/SplashPresenter.java

```diff
@@ -36,6 +36,20 @@ public void refreshData() {
                 settings.edit()
                         .putBoolean("firstRun", false)
                         .apply();
+                //Set to 1 the lastDownload for the taxonomy to be loaded :
+                //This two are needed for all flavors :
+                settings.edit().putLong("lastDownloadcategories", Long.valueOf(1)).apply();
```

Comment:
Issue here: if first run is false ( it's the case for all old users), the settings are not intialized

`if(!settings.contains("lastDownloadcategories")){
             if(!settings.contains("lastDownloadcategories")){
            settings.edit().putLong("lastDownloadcategories", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
            settings.edit().putLong("lastDownloadtags", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
            if (BuildConfig.FLAVOR.equals("off") || BuildConfig.FLAVOR.equals("obf")) {
                settings.edit().putLong("lastDownloadadditives", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
                settings.edit().putLong("lastDownloadcountries", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
                settings.edit().putLong("lastDownloadlabels", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
                //For the moment, ingredients are only need if Diet is use.
                settings.edit().putLong("lastDownloadingredients", Long.valueOf(ProductRepository.TAXONOMY_NOT_TO_BE_LOADED)).apply();
            }
            if (BuildConfig.FLAVOR.equals("off")) {
                settings.edit().putLong("lastDownloadallergens", Long.valueOf(ProductRepository.TAXONOMY_UP_TO_DATE)).apply();
            }
        }`